numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -894,11 +895,11 @@ typedef int (PyArray_FinalizeFunc)(PyArrayObject *, PyObject *);
 #define NPY_ARRAY_IN_ARRAY     (NPY_ARRAY_CARRAY_RO)
 #define NPY_ARRAY_OUT_ARRAY    (NPY_ARRAY_CARRAY)
 #define NPY_ARRAY_INOUT_ARRAY  (NPY_ARRAY_CARRAY | \
-                                NPY_ARRAY_UPDATEIFCOPY)
+                                NPY_ARRAY_WRITEBACKIFCOPY)
```

Comment:
Unfortunately it turns out third party code is using `INOUT_ARRAY` (e.g. `scipy/interpolate/src/_interpolate.cpp`), so I guess we can't just change its value :-(. We could define some new version like `NPY_ARRAY_INOUT_ARRAY2`, or maybe we should just deprecate it and tell people that they should write `NPY_ARRAY_CARRY | NPY_ARRAY_WRITEBACKIFCOPY` if that's what they want.

Reply:
For backward compatibility I prefer the first option. I will extend the warning notice to reflect the possible flags that could trigger it.